### PR TITLE
refactor: apply opinionated api defaults based on accepted adr

### LINKED ISSUE #179: Opinionated API Changes
**Describe the solution you'd like:**
Currently the API is generic. We should consider being more opinionated towards our use case. For example, should the user have to specify a network interface for metadata requests? It may be better that if we see metadata in the request then we automatically creatre a network interface for MMDS requests

**Why do you want this feature:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by jmickey**
**What this PR does / why we need it**:

- Automatically add the metadata service network device
- Default CPU and RAM to 2 and 1024 respectively
- Apply the default namespace is non is provided
- Separate Volumes into RootVolume and AdditionalVolumes

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #179

**Checklist**:

- [x] squashed commits into logical changes
- [x] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: core/models/vmid.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func NewVMID(name, namespace string) (*VMID, error) {
 	}
 
 	if namespace == "" {
-		return nil, coreerrs.ErrNamespaceRequired
+		namespace = "default"
```
Comment (yitsushi) [inline core/models/vmid.go:34]: Can we put store this in a constant, so other packages can use this as value and we have to update only one place?

File: core/models/volumes.go [hunk 2]
```diff
@@ -6,7 +6,10 @@ type Volume struct {
 	ID string `json:"id"`
 	// IsRoot specifies that the volume is to be used as the root volume. A machine
 	// must have a root volume.
-	IsRoot bool `json:"is_root"`
+	//
+	// NOTE: This field does not exist on the gRPC API, it is automatically filled
+	// when converting from the API model to the this internal model.
+	// IsRoot bool `json:"is_root"`
```
Comment (yitsushi) [inline core/models/volumes.go:12]: Is this comment `//` intentional on the Volume model? This is the internal model. Either the comment on the field or the commented out field seems odd.

File: infrastructure/grpc/convert.go [hunk 3]
```diff
@@ -26,6 +26,15 @@ func convertMicroVMToModel(spec *types.MicroVMSpec) (*models.MicroVM, error) {
 			MemoryInMb: int64(spec.MemoryInMb),
 		},
 	}
+
+	if convertedModel.Spec.VCPU == 0 {
+		convertedModel.Spec.VCPU = 2
```
Comment (yitsushi) [inline infrastructure/grpc/convert.go:31]: Same as the namespace, `conts`?

File: infrastructure/grpc/convert.go [hunk 4]
```diff
@@ -26,6 +26,15 @@ func convertMicroVMToModel(spec *types.MicroVMSpec) (*models.MicroVM, error) {
...
+	}
+
+	if convertedModel.Spec.MemoryInMb == 0 {
+		convertedModel.Spec.MemoryInMb = 1024
```
Comment (yitsushi) [inline infrastructure/grpc/convert.go:35]: Same as the namespace and vcpu, `conts`?

**Review by yitsushi — COMMENTED** [review]
I don't know why `gomnd` did not yell on vcpu and memory default values.

File: core/models/volumes.go (hunk 2 above)
Comment (jmickey) [reply core/models/volumes.go:12]: The comment _was_ valid, but I removed the need for `is_root` completely and forget to remove this altogether.

**Review by yitsushi — APPROVED** [review]
Small nit: `MemoryInMb` sounds like it's Megabits not Megabytes, but lgtm

**Pull request merged**